src/common/darktable.h

```diff
@@ -117,13 +117,23 @@ typedef unsigned int u_int;
 #endif
 #endif /* dt_omp_nontemporal */
 
+#define DT_OMP_STRINGIFY(...) #__VA_ARGS__
+#define DT_OMP_PRAGMA(...) _Pragma(DT_OMP_STRINGIFY(omp __VA_ARGS__))
+
 #else /* _OPENMP */
 
 # define omp_get_max_threads() 1
 # define omp_get_thread_num() 0
 
+#define DT_OMP_PRAGMA(...)
+
 #endif /* _OPENMP */
 
+#define DT_OMP_SIMD(clauses) DT_OMP_PRAGMA(simd clauses)
+#define DT_OMP_DECLARE_SIMD(clauses) DT_OMP_PRAGMA(declare simd clauses)
+#define DT_OMP_FOR(clauses) DT_OMP_PRAGMA(parallel for default(firstprivate) schedule(static) clauses)
```

Comment:
@jenshannoschwalm @ralfbrown 
switching to `default(firstprivate)` so it is no longer necessary to explicitly list all variables. This also obviates the need for a separate `DT_OMP_FOR_CLAUSE`.

@LebedevRI imageio_rawspeed.cc seems to pick up `-fopenmp-version=50` from `src-dependencies.cmake`. If it could switch too there would be no need for the special casing.

Reply:
(I'm rather skeptical about deviation from `default(none)`. It was always a feature, not a bug.)

I guess i would suggest to do something along the lines of adding
```
if (USE_OPENMP)
    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR
        CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")

        # Clang has an option to specify the OpenMP standard to use. Specify it.
        set(OPENMP_VERSION_SPECIFIER "-fopenmp-version=51")

        set_property(SOURCE imageio_rawspeed.cc APPEND PROPERTY CMAKE_CXX_FLAGS ${OPENMP_VERSION_SPECIFIER})
    endif()
endif()
```
to `src/imageio/CMakeLists.txt`

Reply:
Not having to explicitly list all of the variables in the OpenMP directive will make the code easier to write and more legible.  The downside (which is what I think Roman is thinking of) is that in certain edge cases, it could introduce hard-to-debug issues when a variable used in the loop should in fact have been declared to be `shared` but was missed.  But that will require a variable which is both mdified by one thread and whose modification needs to be seen by other threads, and where the modification is not to a buffer pointed at by the variable (which will be seen by other threads either way).  I doubt there are more than a couple of such cases in darktable.

I think that on balance, default(firstprivate) is better.